### ISSUE TITLE
Class data derived from TheoryData<SomeType[]> is broken

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#3053]
I still experience this issue in v2.9.2 when using the `ClassData` attribute with a class derived from `TheoryData`. 
See the modified example with the exceptions next to the `Add` calls.
```csharp
    [Theory]
    [ClassData(typeof(ExampleParameters))]
    public void ExampleParameterizedTestMethod(string[] values)
    {
        Assert.All(values, s => Assert.True(s.Length < 10));
    }

    private sealed class ExampleParameters : TheoryData<string[]>
    {
        public ExampleParameters ()
        {
            Add(new string[] { "0" }); //System.ArgumentException: Object of type 'System.String' cannot be converted to type 'System.String[]'.
            Add(new string[] { "0", "2" }); //System.InvalidOperationException: The test method expected 1 parameter value, but 2 parameter values were provided.
            Add(new string[] { "0", "2", "4" }); //System.InvalidOperationException: The test method expected 1 parameter value, but 3 parameter values were provided.
        }
    }
```

_Originally posted by @afuersch in https://github.com/xunit/xunit/issues/3031#issuecomment-2458984932_

**Comment by bradwilson** [comment]
My reply in that other issue:

> Bad news: I can reproduce this.
> 
> Good news: I can't reproduce in v3, so the version of TheoryData in v3 works correctly. The 1.0 release of v3 is coming before year's end (I'm planning to release a final 0.6.0 pre-release shortly, then after a couple weeks of stability testing, release 1.0.)
> 
> I'll look to see whether I can solve this for v2 or not. The difference in design between TheoryData in v2 and v3 doesn't allow me to back-port the v3 code, so that's not an available solution.

**Comment by bradwilson** [comment]
Available in `2.9.3-pre.5` https://xunit.net/docs/using-ci-builds
